src/strings.cpp

```diff
@@ -1713,16 +1720,15 @@ bool LanguagePackHeader::IsValid() const
 bool ReadLanguagePack(const LanguageMetadata *lang)
 {
 	/* Current language pack */
-	size_t len;
-	LanguagePack *lang_pack = (LanguagePack *)ReadFileToMem(lang->file, &len, 1U << 20);
-	if (lang_pack == nullptr) return false;
+	size_t len = 0;
+	std::unique_ptr<LanguagePack> lang_pack(reinterpret_cast<LanguagePack *>(ReadFileToMem(lang->file, len, 1U << 20).release()));
```

Comment:
Yikes.
Perhaps templating `ReadFileToMem` might be worth doing?

Reply:
Looks ugly in the header file, but yeah, might indeed be better.

Reply:
Doesn't really solve the memory allocation problem tough, `LanguagePack` is of undefined size and we still need to obtain a non-array block of memory using `new`.

Reply:
Looking at it again, this is the only use of ReadFileToMem. Perhaps just specialise the function as appropriate?